plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java

```diff
@@ -72,7 +100,23 @@ public Page getNextPage()
             if (projectionsAdapter.isPresent()) {
                 dataPage = projectionsAdapter.get().adaptPage(dataPage);
             }
-            return dataPage;
+            if (dataPage == null) {
+                return null;
+            }
+
+            int positionCount = dataPage.getPositionCount();
+            int[] positionsToKeep = new int[positionCount];
+            try (CloseableIterable<TrinoRow> filteredRows = deleteFilter.filter(CloseableIterable.withNoopClose(TrinoRow.fromPage(columnTypes, dataPage, positionCount)))) {
```

Comment:
```
            try (CloseableIterable<TrinoRow> rows = CloseableIterable.withNoopClose(TrinoRow.fromPage(columnTypes, dataPage, positionCount));
                    CloseableIterable<TrinoRow> filteredRows = deleteFilter.filter(rows)) {
```

Reply:
there is no point to ensure to close `TrinoRow.fromPage`, since it's not a resource.